microcosm_postgres/factories/shards.py

```diff
@@ -128,26 +118,14 @@ def select_shard():
             return graph.client_shard.get(client_id, GLOBAL_SHARD_NAME)
         return GLOBAL_SHARD_NAME
 
-    def shard_chooser(mapper, instance, clause=None):
-        """Choose which shard to send the object instance to for mutations"""
-        return select_shard()
+    class SingleShardedSession(Session):
+        """
+        A session that always uses the same shard.
 
-    def execute_chooser(context):
-        """Choose which shard to query."""
-        return [select_shard()]
+        This is useful for read-only sessions.
+        """
 
-    def identity_chooser(mapper, primary_key, *, lazy_loaded_from, **kw):
-        """Choose which shard to select the server generated id from."""
-        return [select_shard()]
+        def get_bind(self, *args, **kwargs):
+            return graph.shards[select_shard()]
```

Comment:
This is the main difference, normal session with a single bind, not the weird multi bind ShardedSession that's harder to work with